pkg/pub_package_reader/lib/pub_package_reader.dart

```diff
@@ -307,6 +308,36 @@ Iterable<ArchiveIssue> checkAuthors(String pubspecContent) sync* {
   }
 }
 
+final _preDart3 = VersionConstraint.parse('<3.0.0');
+final _firstDart3Pre = Version.parse('3.0.0').firstPreRelease;
+
+/// Checks if the version range is acceptable by current SDKs.
+Iterable<ArchiveIssue> checkSdkVersionRange(Pubspec pubspec) sync* {
+  final sdk = pubspec.environment?['sdk'];
+  if (sdk == null ||
+      sdk.isAny ||
+      sdk is! VersionRange ||
+      sdk.min == null ||
+      sdk.min!.isAny ||
+      sdk.max == null ||
+      sdk.max!.isAny) {
+    yield ArchiveIssue(
+        'Dart SDK constraint with min and max range must be specified.');
+    return;
+  }
+
+  // No known version exists.
+  if (sdk.intersect(_preDart3).isEmpty) {
+    yield ArchiveIssue('SDK constraint is outside of current Dart SDKs.');
```

Comment:
Can we improve the message to say: SDK constraint does not allow any known Dart SDK.